src/state_machine/coordinator/mod.rs

```diff
@@ -972,4 +973,71 @@ pub mod test {
 
         assert_eq!(signers, loaded_signers);
     }
+
+    pub fn gen_nonces<Coordinator: CoordinatorTrait, SignerType: SignerTrait>(
+        num_signers: u32,
+        keys_per_signer: u32,
+    ) {
```

Comment:
I know you now what this test does, but I think that we should add a bit more documentation to this test. It's not clear what is being tested an why the outcome is the correct one. I mean, I know what's going on but someone who is being onboarded here might have a tough time figuring a lot of this out. So I'd add more detailed comments during the test and add a description above the test function as well.

Reply:
Done